### PR TITLE
docs(rust): complete intra-docs in api documentation

### LINKED ISSUE #10904: Intra-doc links in rust api documentation
### Problem description

I think it would be a good idea to provide direct links to rustdoc pages of structs, traits etc. As is, when you read documentation and want to read up on, let's say, `DataFrame` you need to search the whole doc or manually navigate to frame module. It would increase quality of life if you could click an element and be redirected to a corresponding doc page (in this case [DataFrame](https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html)).

At the time of writing the issue:
image
Proposal:
image

If such a change is desired I could work on it.

### PR DESCRIPTION
**Pull request by mrogowski11**
Resolves #10904 

The commit includes several new intra-doc elements within the Rust API documentation and addresses a few pre-existing, malfunctioning ones.

### CODE REVIEW

File: crates/polars/src/docs/eager.rs [hunk 1]
```diff
@@ -120,7 +124,7 @@
 //! let multiplied = ca_int * 2.0;
 //! let multiplied = s_flt * 2.0;
 //!
-//! // or broadcast Series to match the operands type
+//! // or broadcast ['Series'] to match the operands type
```
Comment (orlp) [inline crates/polars/src/docs/eager.rs:127]: That don't look right :)

**Review by orlp — CHANGES_REQUESTED** [review]

**Comment by orlp** [comment]
I quickly skimmed it looks good, thanks! Could you quickly run [`cargo deadlinks`](https://github.com/deadlinks/cargo-deadlinks) just to be sure all of them actually work?

**Comment by mrogowski11** [comment]
deadlinks returned a couple of dead legacy links which where ommited by rustdoc. Fixed them in:
https://github.com/pola-rs/polars/pull/11007/commits/1bfa69421f1a57fca102694940d9642f72793ba3

With that change, no internal errors are shown. Keep in mind that it seems that deadlinks lists links from external crates when run on a whole project (mainly it shows warnings for [dyn-clone](https://crates.io/crates/dyn-clone) and [equivalent](https://crates.io/crates/dyn-clone)). deadlinks was run with _--no-build_ to force _--all-features_ on cargo doc.
[deadlinks.log](https://github.com/pola-rs/polars/files/12561962/deadlinks.log)

File: crates/polars/src/docs/eager.rs (hunk 1 above)
Comment (mrogowski11) [reply crates/polars/src/docs/eager.rs]: Good catch. Reverted in:
https://github.com/pola-rs/polars/pull/11007/commits/1bfa69421f1a57fca102694940d9642f72793ba3#diff-a976127b0782570471eeed4567ed71a313680122e0ae84a1c1ebafa3147c230a

**Review by stinodego — APPROVED** [review]
This is great! I say we merge this before it gathers any conflicts :) we can always fine-tune later.

**Review by orlp — APPROVED** [review]

**Pull request merged**